core/templates/dev/head/profile/Preferences.js

```diff
@@ -35,6 +36,46 @@ oppia.controller('Preferences', ['$scope', '$http', '$rootScope', '$modal', '$ti
     _saveDataItem('user_bio', userBio);
   };
 
+  $scope.changedAtLeastOnce = false;
+  $scope.subjectInterestsWarningText = null;
+  $scope.TAG_REGEX = '^[a-z ]+$';
+
+  $scope.updateSubjectInterestsWarning = function(subjectInterests) {
+    var alpha = /^[A-Za-z]+$/;
+
+    if (subjectInterests instanceof Array) {
+      for (var i=0; i<subjectInterests.length; i++) {
+        if (typeof subjectInterests[i] === 'string') {
+          console.log(subjectInterests[i]);
+          if (alpha.test(subjectInterests[i])) {
+            if ($scope.subjectInterestsWarningText == null) {
+              $scope.subjectInterestsWarningText = '';
+            }
+          } else {
+            $scope.subjectInterestsWarningText = (
+            'Subject interests can only have alphabetic characters.');
+          }
+        } else {
+          $scope.subjectInterestsWarningText = (
```

Comment:
Here, I would just throw an error instead:

    console.error('Error: received bad value for subject interests. Expected list of strings, got ', subjectInterests);
    throw Error('Error: received bad value for subject interests.');

Ditto below. Reasoning: if the user sees this error, it's totally unclear how they would fix it. This sort of thing would indicate a problem at the system level and we want to make the error as obvious as possible so that the developer catches it.